src/server/game/Entities/Player/Player.cpp

```diff
@@ -2665,6 +2667,31 @@ void Player::GiveLevel(uint8 level)
     sScriptMgr->OnPlayerLevelChanged(this, oldLevel);
 }
 
+uint8 Player::GetExpansionLevelLimit() const
+{
+    uint8 max_lvl = GetMaxLevelForExpansion(_acc_expansion);
+
+    // if _acc_expansion stores some undefined value, GetMaxLevelForExpansion
+    // returns 0 and it should be treated like the latest expansion
+    // with deafult max lvl
+    return max_lvl != 0 ? max_lvl : DEFAULT_MAX_LEVEL;
+}
+
+bool Player::IsMaxLevel() const
```

Comment:
Replace the entire function body with just `return GetLevel() >= GetUInt32Value(PLAYER_FIELD_MAX_LEVEL);`

Reply:
> change PLAYER_FIELD_MAX_LEVEL to GetMaxLevelForExpansion(GetSession()->Expansion())

I like the idea of this variable as it solves the reason why I added `uint8 _acc_expansion` to exclude extra function calls. 
But it won't work simply like that as it requires checks between `GetMaxLevelForExpansion` and `MaxPlayerLevel` in the config.

Reply:
Avoiding extra non virtual function calls that simply read one field on a non hot code path?